src/main/java/org/craftercms/studio/impl/v2/security/listener/AbstractAuditListener.java

```diff
@@ -74,4 +75,32 @@ protected void recordAuthenticationEvent(String operation, AbstractAuthenticatio
             // This never happens
         }
     }
+
+    /**
+     * Perform audit log for session timeout event
+     * @param operation audit type operation
+     * @param context the security context of the timeout session
+     * @param message a log message for the event
+     */
+    protected void recordSessionTimeoutEvent(String operation, SecurityContext context, String message) {
+        try {
+            var name = context.getAuthentication().getName();
+            var username = StringUtils.substring(name, 0, MAX_USERNAME_LENGTH);
+            var site = siteService.getSite(studioConfiguration.getProperty(CONFIGURATION_GLOBAL_SYSTEM_SITE));
+            AuditLog auditLog = auditServiceInternal.createAuditLogEntry();
+            auditLog.setOperation(operation);
+            auditLog.setActorId(username);
+            auditLog.setSiteId(site.getId());
+            auditLog.setPrimaryTargetId(username);
+            auditLog.setPrimaryTargetType(TARGET_TYPE_USER);
+            auditLog.setPrimaryTargetValue(username);
+            auditServiceInternal.insertAuditLog(auditLog);
+
+            if (isNotEmpty(message)) {
+                logger.info(message, name);
+            }
+        } catch (SiteNotFoundException e) {
+            // This never happens
```

Comment:
Nonetheless, let's log the error as behavior might silently change in the future

Reply:
Updated